src/python_testing/TC_DeviceConformance.py

```diff
@@ -50,6 +50,22 @@ async def setup_class_helper(self):
         self.xml_device_types, problems = build_xml_device_types()
         self.problems.extend(problems)
 
+    def _get_device_type_id(self, device_type_name: str) -> int:
+        id = [id for id, dt in self.xml_device_types.items() if dt.name.lower() == device_type_name.lower()]
+        if len(id) != 1:
+            self.fail_current_test(f"Unable to find {device_type_name} device type")
+        return id[0]
+
+    def _has_nim(self):
+        nim_id = self._get_device_type_id('network infrastructure manager')
```

Comment:
For `_get_device_type_id` handle case normalization? Would the call `_get_device_type_id('Network Infrastructure Manager')` also work?

Reply:
yup